nautobot/extras/api/views.py

```diff
@@ -198,6 +199,21 @@ class CustomLinkViewSet(ModelViewSet):
     filterset_class = filters.CustomLinkFilterSet
 
 
+#
+# Dynamic Groups
+#
+
+
+class DynamicGroupViewSet(ModelViewSet):
+    """
+    Manage Dynamic Groups through DELETE, GET, POST, PUT, and PATCH requests.
+    """
+
+    queryset = DynamicGroup.objects.prefetch_related("content_type")
+    serializer_class = serializers.DynamicGroupSerializer
+    filterset_class = filters.DynamicGroupFilterSet
+
```

Comment:
This might be tricky at least from a Swagger schema perspective since decorating the view method with `@swagger_auto_schema` requires a `request_body=` argument with an object serializer. This might be ugly.